<?xml version="1.0"?>
<launch>
  <arg name="map_frame" default="/map" />
  <arg name="twist_topic" default="/cmd_vel" />
  <arg name="path_topic" default="/waypoints_raw" />
  <arg name="current_pose_topic" default="/current_pose" />
  <arg name="opt_path_topic" default="/dwa/opt_path_points" />
  <arg name="target_marker_topic" default="/dwa/target_point" />
  <arg name="target_search_interval" default="10" />
  <arg name="max_linear_vel" default="0.5" />
  <arg name="min_linear_vel" default="0.05" />
  <arg name="max_angular_vel" default="3.0" />
  <arg name="min_angular_vel" default="-3.0" />
  <arg name="lookahead_distance" default="0.5" />
  <arg name="weight_obs" default="0.4" />
  <arg name="weight_vel" default="0.2" />
  <arg name="weight_angle" default="0.4" />
  <arg name="obs_avoid_threshold" default="0.6" />

  <node name="dwa_node" pkg="dwa" type="dwa_node" output="screen" respawn="true" >
    <param name="map_frame" value="$(arg map_frame)" />
    <param name="twist_topic" value="$(arg twist_topic)" />
    <param name="path_topic" value="$(arg path_topic)" />
    <param name="current_pose_topic" value="$(arg current_pose_topic)" />
    <param name="opt_path_topic" value="$(arg opt_path_topic)" />
    <param name="target_marker_topic" value="$(arg target_marker_topic)" />
    <param name="target_search_interval" value="$(arg target_search_interval)" />
    <param name="max_linear_vel" value="$(arg max_linear_vel)" />
    <param name="min_linear_vel" value="$(arg min_linear_vel)" />
    <param name="max_angular_vel" value="$(arg max_angular_vel)" />
    <param name="min_angular_vel" value="$(arg min_angular_vel)" />
    <param name="lookahead_distance" value="$(arg lookahead_distance)" />
    <param name="weight_obs" value="$(arg weight_obs)" />
    <param name="weight_vel" value="$(arg weight_vel)" />
    <param name="weight_angle" value="$(arg weight_angle)" />
    <param name="obs_avoid_threshold" value="$(arg obs_avoid_threshold)" />
  </node>
</launch>
